### PR TITLE
Add script(s) for issue 2399

### PR DESCRIPTION
**Pull request by VNikiforov-dev**
ATF Test Scripts to check #[2399]

This PR is **ready for Internal** review.

### Summary
Scripts to check 2399 issue

### ATF version
develop

### Changelog
- Added scripts to check 2399 issue

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---------------------------------------------------------------------------------------------------
```
Comment (aderiabin) [inline test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua:1]: @VNikiforov-dev Please add this script to `test_sets/Defects/Defects_release_8_0.txt` test set

File: test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+-- User story: https://github.com/smartdevicelink/sdl_core/issues/2399
+---------------------------------------------------------------------------------------------------
+-- Description:
+-- SDL does respond UNSUPPORTED_RESOURCE to SystemRequest(QUERY_APPS) in case app is registered with protocol version
+-- less than 4.0
+---------------------------------------------------------------------------------------------------
+-- In case:
+-- 1) App is registered and activated
```
Comment (aderiabin) [inline test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua:9]: @VNikiforov-dev I would propose to use numeration with dots for test steps
``` 1) -> 1. ```

File: test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+-- 2) Mobile app requests SystemRequest(QUERY_APPS)
```
Comment (aderiabin) [inline test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua:10]: @VNikiforov-dev Please use one of `App` or `Mobile app` in bound of whole script

File: test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local common = require('user_modules/sequences/actions')
+local runner = require('user_modules/script_runner')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local protocol_version = 3
```
Comment (aderiabin) [inline test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua:22]: @VNikiforov-dev I would propose to use `camelCase` naming style instead `snake_case` here
```
local protocol_version = 3
=>
local protocolVersion = 3
```
And rename this variable according its purpose 
```
protocolVersion => maxSdlProtocolVersion
```

**Review by aderiabin — APPROVED** [review]

File: test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Update INI File", common.setSDLIniParameter, {"MaxSupportedProtocolVersion", protocol_version})
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("Register App", common.registerApp)
+
+runner.Title("Test")
+runner.Step("SystemRequest (QUERY_APPS) request", systemRequest)
+
+runner.Title("Postconditions")
+runner.Step("Stop SDL", common.postconditions)
```
Comment (dboltovskyi) [inline test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua:41]: Please add script to test set for issues of 8.0: https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_sets/Defects/Defects_release_8_0.txt

File: test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua [hunk 6]
```diff
@@ -0,0 +1,41 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/smartdevicelink/sdl_core/issues/2399
```
Comment (dboltovskyi) [inline test_scripts/Defects/8_0/2399__SDL_respond_UNSUPPORTED_RESOURCE_in_case_SDL_4_0_was_omited.lua:2]: Replace `User story` by `Issue`

**Review by dboltovskyi — CHANGES_REQUESTED** [review]

**Review by dboltovskyi — APPROVED** [review]

**Review by VjKlepikov — APPROVED** [review]

**Pull request closed without merging**